tests/framework/statistics/function.py

```diff
@@ -3,155 +3,162 @@
 
 """Module for statistical functions."""
 
-
 from abc import ABC, abstractmethod
-from numbers import Number
-from typing import Any, List
+from typing import Any
 # pylint: disable=E0611
 from statistics import mean, stdev
+from pydoc import locate
+
+
+class FunctionFactory:
+    """Function factory class."""
+
+    @classmethod
+    def get(cls, func_cls_name) -> 'Function':
+        """`func_cls_name` must be a valid function class name."""
+        return locate(f"framework.statistics.function.{func_cls_name}")
 
 
 # pylint: disable=R0903
-class StatisticFunction(ABC):
+class Function(ABC):
     """Statistic function abstract class."""
 
-    def __init__(self, results: List[Number]):
+    def __init__(self, name):
         """Initialize the statistic function."""
-        self.results = results
+        self._name = name
 
     @abstractmethod
-    def __call__(self) -> Any:
+    def __call__(self, result: Any) -> Any:
         """Builtin function needs to be implemented."""
 
-    @classmethod
-    @abstractmethod
-    def name(cls) -> str:
-        """Return the name identifier for the class."""
+    @property
+    def name(self) -> str:
+        """Return the a name identifier for the class."""
+        return self._name
 
 
 # pylint: disable=R0903
-class GetFirstObservation(StatisticFunction):
-    """A function which return the first observation."""
+class ValuePlaceholder(Function):
```

Comment:
I had this in picture once but I think that using interchangeably measurements and statistics for the corner case you describe will bring more confusion to the table. Statistics and measurements have defining roles in this framework and helps grasping the resources needed by a successful statistical exercise. Statistics refer to insights over data, while the measurements refer to the numerical quantitation of the attributes of an object. 

Putting all of this together will lead us to insights (statistics) over the attributes (measurement name) to numerical quantitation (measurement unit) of an object (network, block, vsock). I find every piece defined previously very important for concrete expectations throughout the test design and format of the test result.

Regarding the fact we have to run 1 iteration for getting statistics data, this is not necessary the case (it is for existing tests, but we need to keep room for other scenarios) - e.g a tool produces for 10 iterations measurement data and also in the 10th iteration tells you the average CPU utilization for the previous 10 iterations.

Let me know in case your questions were not answered.

Reply:
Added documentation for this function. Please take a look.